pkg/minibroker/mysql.go

```diff
@@ -34,53 +45,41 @@ func (p MySQLProvider) Bind(services []corev1.Service, params map[string]interfa
 
 	host := buildHostFromService(service)
 
-	database := ""
-	dbVal, ok := params["mysqlDatabase"]
-	if ok {
-		database, ok = dbVal.(string)
-		if !ok {
-			return nil, errors.Errorf("mysqlDatabase not a string")
-		}
-	}
-
-	var user, password string
-	userVal, ok := params["mysqlUser"]
-	if ok {
-		user, ok = userVal.(string)
-		if !ok {
-			return nil, errors.Errorf("mysqlUser not a string")
-		}
+	database := provisionParams.DigStringOr("mysqlDatabase", "")
+	user := provisionParams.DigStringOr("mysqlUser", rootMysqlUsername)
 
-		passwordVal, ok := chartSecrets["mysql-password"]
-		if !ok {
-			return nil, errors.Errorf("mysql-password not found in secret keys")
-		}
-		password, ok = passwordVal.(string)
-		if !ok {
-			return nil, errors.Errorf("password not a string")
-		}
+	var passwordKey string
+	if user == rootMysqlUsername {
+		passwordKey = "mysql-root-password"
```

Comment:
(This and similar) shouldn't that be a `const`?

Reply:
I started to move things to consts and the readability became really bad. I'll stick to the current approach.